hypertrace-gradle-docker-plugin/src/main/java/org/hypertrace/gradle/docker/DockerPlugin.java

```diff
@@ -86,15 +86,27 @@ private void addDockerBuildTaskForImage(
   }
 
   private void addDefaultTags(Project project, DockerPluginExtension extension) {
-    extension.tag(this.getVersionString(project));
-    extension.tag(DockerTag.LATEST, tag -> tag.onlyIf(unused -> extension.tagLatest.get()));
+    if (this.isReleaseVersion(project)) {
+      extension.tag(this.getVersionString(project));
+      extension.tag(DockerTag.LATEST, tag -> tag.onlyIf(unused -> extension.tagLatest.get()));
+    }
+    extension.tag(this.getBranchTag(project));
   }
 
   private String getVersionString(Project project) {
     return project.getVersion()
                   .toString();
   }
 
+  private boolean isReleaseVersion(Project project) {
+    return !this.getVersionString(project)
+                .contains("SNAPSHOT");
+  }
+
+  private String getBranchTag(Project project) {
+    return getEnvironmentVariable("CIRCLE_BRANCH").orElse("local");
```

Comment:
if I understand  correctly this "local" will be the image tag when run from normal checkout? while a bit of a bikeshed I prefer "test" if there isnt a strong opinion.

Reply:
maybe add a comment that this will result in "master" tag on normal CI and "local" ( or "test" ) otherwise as I think that is the intent but might not be obvious.

Reply:
Correct - I was going to use the branch name, but it didn't seem worth introducing a git dependency/complexity for the dev build case. No strong opinions on the specific label, so I'll switch it to "test" and add a comment.